transcendence-app/src/chat/chat.gateway.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  ClassSerializerInterceptor,
+  Logger,
+  UseGuards,
+  UseInterceptors,
+} from '@nestjs/common';
+import {
+  ConnectedSocket,
+  MessageBody,
+  OnGatewayConnection,
+  OnGatewayDisconnect,
+  OnGatewayInit,
+  SubscribeMessage,
+  WebSocketGateway,
+  WebSocketServer,
+} from '@nestjs/websockets';
+import { Socket, Server } from 'socket.io';
+import { v4 as uuidv4 } from 'uuid';
+import { WsAuthenticatedGuard } from '../shared/guards/ws-authenticated.guard';
+import { User } from '../user/user.domain';
+import { ChatService } from './chat.service';
+
+type Message = {
+  id: string;
+  user: User;
+  content: string;
+  createdAt: number;
+};
+
+@WebSocketGateway({ path: '/api/v1/socket.io' })
+@UseGuards(WsAuthenticatedGuard)
+@UseInterceptors(ClassSerializerInterceptor)
+export class ChatGateway
+  implements OnGatewayInit, OnGatewayConnection, OnGatewayDisconnect
+{
+  @WebSocketServer() server!: Server;
+
+  logger = new Logger(ChatGateway.name);
+  messages = new Set<Message>();
+  connectedUsers = new Map<string, User>();
```

Comment:
I don't understand why do we need this to have both the userId, and the User (that already contains the user id)

Reply:
Good point, thanks! I've changed this to a `Set` of user IDs.